packages/backend-proxy-middleware/src/base/proxy.ts

```diff
@@ -332,7 +332,7 @@ export async function generateProxyMiddlewareOptions(
     }
 
     backend.proxy = getCorporateProxyServer(backend.proxy);
-    if (backend.proxy && !isHostExcludedFromProxy(proxyOptions.target)) {
+    if (backend.proxy && !isHostExcludedFromProxy(proxyOptions.target!)) {
```

Comment:
Looking at this again, after responding to @devinea , I think this is incorrect. It is not guaranteed that target is defined.

Reply:
Yup, I missed that `getDestinationUrlForAppStudio` can return `undefined`.

Reply:
But if that happens, the proxy cannot work, correct? So if proxyOptions.target is undefined at that point in time we should throw an error.

Reply:
Yes, without a `target` the proxy can't work.

Reply:
Done with https://github.com/SAP/open-ux-tools/pull/581/commits/4ad8951f302d05ab47b993d22f5b2d33fbffcb73